SU2_CFD/include/numerics_structure.hpp

```diff
@@ -821,6 +827,25 @@ class CNumerics {
                           su2double val_thermal_conductivity,
                           su2double val_heat_capacity_cp);
     
+  /*!
+   * \brief Compute the projection of the viscous fluxes into a direction for general fluid model.
+   * \param[in] val_primvar - Primitive variables.
+   * \param[in] val_gradprimvar - Gradient of the primitive variables.
+   * \param[in] val_turb_ke - Turbulent kinetic energy
+   * \param[in] val_normal - Normal vector, the norm of the vector is the area of the face.
+   * \param[in] val_laminar_viscosity - Laminar viscosity.
+   * \param[in] val_eddy_viscosity - Eddy viscosity.
+   * \param[in] val_reynolds_stress - Reynolds stress.
+   * \param[in] val_perturbed_rsm - Perturbed Reynolds Stress Matrix
+   */
+
+  void GetViscousProjFlux(su2double *val_primvar, su2double **val_gradprimvar,
```

Comment:
I think this routine is overloaded multiple times.. any unexpected use cases that might break things? For example, if someone tries to run the UQ module with a non-ideal fluid model?

Reply:
That's a good point. I tweaked the implementation and removed this overloaded function. Now it uses the two existing GetViscousProjFlux functions. This was possible by incorporating one of the other suggestions; the one about moving some of the data structures to the CNumerics class and initializing them in the constructor. 

This way, the Perturbed Reynolds Stress Tensor is a variable in the CNumerics class and doesn't need to be passed into the GetViscousProjFlux function. 

The UQ methodology is now implemented for the general fluid, and for both the CorrectedGrad and AvgGrad classes as well.